activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/service/ModelService.java

```diff
@@ -210,23 +228,108 @@ public Model updateModelContent(Model modelToBeUpdate,
-                                                     modelContentService.getModelContentId(model)))
-                .map(content -> new FileContent(fileContent.getFilename(),
-                                                fileContent.getContentType(),
-                                                content))
-                .orElse(fileContent);
+
+  public FileContent checkAndFixModelContent(Model model, FileContent fileContent) {
+      FileContent fixedFileContent = null;
+      switch (model.getType()){
+        case "PROCESS": {
+          fixedFileContent = this.createFixedProcessFileContent(model, fileContent);
+          break;
+        }
+        case "FORM":{
+          fixedFileContent = this.createFixedFormModelFileContent(model, fileContent);
+          break;
+        }
+        default:
+          fixedFileContent = this.createFixedUIModelFileContent(model, fileContent);
+      }
+      return fixedFileContent;
+  }
+
+  private FileContent createFixedProcessFileContent(Model model, FileContent fileContent) {
+    ModelContent modelContent = this.createBpmnModelContentFromModel(model, fileContent);
+    this.fixProcessModel((BpmnProcessModelContent) modelContent);
+    return new FileContent(fileContent.getFilename(), fileContent.getContentType(),
+      this.convertModelContentToFile(modelContent, model));
+  }
+
+  private FileContent createFixedUIModelFileContent(Model model, FileContent fileContent) {
+    try {
+      ObjectNode jsonNode = (ObjectNode) objectMapper.readTree(fileContent.getFileContent());
+      String actualId = this.modelIdentifiers.get(jsonNode.get("id").asText());
+      if(actualId != null) {
+        jsonNode.put("id", actualId);
+      }
+      return new FileContent(fileContent.getFilename(), fileContent.getContentType(), objectMapper.writeValueAsBytes(jsonNode));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private FileContent createFixedFormModelFileContent(Model model, FileContent fileContent) {
+    try {
+      JsonNode jsonNode = objectMapper.readTree(fileContent.getFileContent());
+      ObjectNode jFormRepresentation = (ObjectNode) jsonNode.get("formRepresentation");
+      String actualId = this.modelIdentifiers.get(jFormRepresentation.get("id").asText());
+      if(actualId != null) {
+        jFormRepresentation.put("id", actualId);
+      }
+      return new FileContent(fileContent.getFilename(), fileContent.getContentType(), objectMapper.writeValueAsBytes(jsonNode));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
     }
+  }
+
+  private void fixProcessModel(BpmnProcessModelContent processModelContent){
+    processModelContent.getBpmnModel().getProcesses().forEach(process -> {
+      String validIdentifier = this.modelIdentifiers.get(process.getId());
+      if(validIdentifier != null && validIdentifier != process.getId()){
+        process.setId(validIdentifier);
+      }
+      process.getFlowElements().stream()
```

Comment:
would not it look simpler using a normal for each loop in this case?

Reply:
in this case i'm taking advantage of the filter so i would just handle the CallActivity elements that must be fixed.